File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> mpp;
        int left=0,ans=0;
        for(int right=0;right<s.size();right++){
            mpp[s[right]]++;
            while(mpp[s[right]]>1)
                mpp[s[left++]]--;
            ans=max(ans,right-left+1);
        }
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
#from collections import DefaultDict
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        mpp={}
        left,ans=0,0
        for right in range(len(s)):
            mpp[s[right]]=mpp.get(s[right],0)+1
            while mpp[s[right]]>1:
                mpp[s[left]]-=1
                if mpp[s[left]]==0:
                    del mpp[s[left]]
                left+=1
            ans=max(ans,right-left+1)

        return ans